src/image/Image.cc

```diff
@@ -214,14 +233,35 @@ int Image::insert(SqlDB *db, string& error_str)
             }
             // else format in the IMAGE template
         }
-        else
+        else //CDROM, KERNEL, RAMDISK, CONTEXT
         {
-            // It's filled by the driver depending on the type of file.
-            format = "";
+            error_str = "No PATH nor SOURCE in template.";
+            goto error_common;
         }
     }
 
-    state = LOCKED; //LOCKED till the ImageManager copies it to the Repository
+    // -------------------------------------------------------------------------
+    // State is LOCKED till the ImageManager copies it to the Repository.
+    // Backup Images set to READY as it is already in the repo
+    // -------------------------------------------------------------------------
+    state = LOCKED;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that the presence of non-empty statements like `a + b` without clear context or purpose within the codebase can lead to potential confusion or misinterpretation. To mitigate this issue, it&#39;s advisable to ensure that such statements are appropriately integrated into larger constructs that contribute meaningfully to the program&#39;s logic or functionality. This could involve assigning the result of the expression to a variable.